lib/order.rb

```diff
@@ -0,0 +1,67 @@
+require_relative "customer.rb"
+require "csv"
+
+class Order
+  attr_accessor :products, :customer, :fulfillment_status
+  attr_reader :id
+
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    @id = id
+    @products = products
+    @customer = customer
+    @fulfillment_status = fulfillment_status
+    until [:pending, :paid, :processing, :shipped, :complete].include?(fulfillment_status)
+      raise ArgumentError, "Please enter a valid status."
+    end
+  end
+
+  def total
+    array_of_prices = @products.map do |_, price|
+      price
+    end
+    tax = array_of_prices.sum * 0.075
+    return (array_of_prices.sum + tax).round(2)
+  end
+
+  def add_product(name, price)
+    if @products.keys.include?(name.downcase)
+      raise ArgumentError, "This product is already in your order."
+    else
+      @products[name] = price
+    end
+  end
+
+  def remove_product(name, price)
+    if !(@products.keys.include?(name.downcase))
+      raise ArgumentError, "#{name} is not in your order."
+    end
+    @products.delete(name.downcase)
+  end
+
+  def self.hash(products)
+    products_hash = {}
+    split_products = products.split(";")
+    split_products.each do |pair_product|
```

Comment:
1) Good work getting this tricky order-parsing logic figured out
2) I love the idea of breaking this out into a separate method - good organization
3) The name of this method, `hash`, doesn't give the reader much information about what it does. Something like `parse_product_list` might be a stronger choice.